### PR TITLE
Release PR 1.5.0

### PR DESCRIPTION
**Pull request by apeltzer**
Adding here "just" the points addressed in this release 👇🏻 

### `Added`

- [#202](https://github.com/nf-core/demultiplex/pull/202) Added cellranger mkfastq subworkflow for demultiplexing 10x samples.
- [#206](https://github.com/nf-core/demultiplex/pull/206) Add test with uncompressed data.
- [#208](https://github.com/nf-core/demultiplex/pull/208) Added parameter for removing adapter information from samplesheets.
- [#210](https://github.com/nf-core/demultiplex/pull/210) Add checkqc module.
- [#212](https://github.com/nf-core/demultiplex/pull/212) Added resource setting arguments to mkfastq module to work with github CI.
- [#214](https://github.com/nf-core/demultiplex/pull/214) Added test_pe (paired end) profile.
- [#220](https://github.com/nf-core/demultiplex/pull/220) Added kraken2.
- [#221](https://github.com/nf-core/demultiplex/pull/221) Added checkqc_config to pipeline schema.
- [#225](https://github.com/nf-core/demultiplex/pull/225) Added test profile for multi-lane samples, updated handling of such samples and adapter trimming.
- [#234](https://github.com/nf-core/demultiplex/pull/234) Added module for samplesheet validation.
- [#236](https://github.com/nf-core/demultiplex/pull/236) Add samplesheet generation.

### `Changed`

- [#204](https://github.com/nf-core/demultiplex/pull/204) Update to latest bcl_demultiplex sub workflow.
- [#210](https://github.com/nf-core/demultiplex/pull/210) Update bcl2fastq and bcl_demultiplex.
- [#214](https://github.com/nf-core/demultiplex/pull/214) Updated method for removing adapters from samplesheet, added custom AdapterRemover function.
- [#210](https://github.com/nf-core/demultiplex/pull/212) Update bcl2fastq and bcl_demultiplex.
- [#216](https://github.com/nf-core/demultiplex/pull/216) List fastq reports for R1 and R2 separately in multiqc report.
- [#219](https://github.com/nf-core/demultiplex/pull/219) Modified workflow to store samplesheet in results folder.
- [#217](https://github.com/nf-core/demultiplex/pull/217) Update all nf-core modules and tests.
- [#240](https://github.com/nf-core/demultiplex/pull/240) Updated samshee, stub script and error strategy

### `Fixed`

- [#224](https://github.com/nf-core/demultiplex/pull/217) Fix input filename collision.
- [#231](https://github.com/nf-core/demultiplex/pull/231) Fix checkqc error message and .command.log emission.

### CODE REVIEW

**Comment by nschcolnicov** [comment]
We are looking into this issue with @atrigila.
For the utils_nfvalidation_plugin, the issue seems to be with where the output was expected to be located when writing the nf-tests. 
The utils_nfvalidation_plugin/main.nf uses this line to print the help message:
image
This gets printed into the nextflow.log, but not into stdout:
image

image

One alternative to fix this would be to replace the log.info with a println, this way the output will be redirected to stdout insted of to the nextflow.log:
image


@apeltzer An alternative would be to re write the nf-tests so that they check the nextflow.log, or something similar. What do you think?

**Comment by grst** [comment]
No strong opinion... I think it's fine (or even preferable) if help is written to STDOUT (most CLI tools do that). 
Have you checked what other pipelines do?

**Comment by apeltzer** [comment]
Variantbenchmarking  / taxprofiler both use the unmodified utils_nfvalidation_plugin/main.nf

**Comment by nschcolnicov** [comment]
@apeltzer @grst I think using log instead of println is better, the log message sticks a little longer in the terminal than println, making it easier to read, besides that, you get it printed in a different color which also helps visibility, but you can do that with println too, so not the biggest advantage. 
I'm not sure why the nf-tests were passing before for this subworkflow, could it be that they weren't being properly executed? I'll create an issue in the modules repo

**Comment by apeltzer** [comment]
OK - lets stick with `nf-validation` for now, the bug (or whatever it in the end turns out to be) was reported upstream and is probably tackled when nf-core moves towards wider usage of nf-schema in favor of nf-validation, thus also updating the required subworkflows / replace them entirely.

**Comment by nschcolnicov** [comment]
@apeltzer @atrigila @grst Removed workflow testing from CI to avoid running the tests on the subworkflows, because UTILS_NEXTFLOW_PIPELINE and UTILS_NFVALIDATION_PLUGIN have nf-tests that are not working. The reason for them failing has been documented and an issue was opened: https://github.com/nf-core/modules/issues/6166.
These failing nf-tests don't seem to have any impact on the disambiguate pipeline, so they have been disabled in this PR, to prepare the dev branch for release.
We will update the CI tests once the issue with the subworfklows is sorted out.

**Comment by apeltzer** [comment]
So its just the downloading feature failing now - filed an issue for that one: https://github.com/nf-core/tools/issues/3111

File: modules/nf-core/custom/dumpsoftwareversions/templates/dumpsoftwareversions.py [hunk 1]
```diff
@@ -3,7 +3,6 @@
```
Comment (nschcolnicov) [inline modules/nf-core/custom/dumpsoftwareversions/templates/dumpsoftwareversions.py:3]: CUSTOM_DUMPSOFTWAREVERSIONS is no longer being used in this pipeline

File: nextflow.config [hunk 2]
```diff
@@ -27,15 +43,16 @@ params {
...
+    outdir                       = null
+    publish_dir_mode             = 'copy'
+    email                        = null
+    email_on_fail                = null
+    plaintext_email              = false
+    monochrome_logs              = false
+    hook_url                     = null
+    help                         = false
+    version                      = false
+    pipelines_testdata_base_path = 'https://raw.githubusercontent.com/nf-core/test-datasets/'
```
Comment (nschcolnicov) [inline nextflow.config:55]: this parameter is not used anywhere

File: nextflow_schema.json [hunk 3]
```diff
@@ -124,6 +146,20 @@
                 }
             }
         },
+        "checkqc_options": {
+            "title": "CheckQC options",
+            "type": "object",
+            "description": "",
+            "default": "",
+            "properties": {
+                "checkqc_config": {
```
Comment (nschcolnicov) [inline nextflow_schema.json:155]: this parameter is not defined in the nextflow config, should it be? Also, if the default is a null value, shouldn't we remove the line from the schema as we do for other null default values?

File: nextflow_schema.json [hunk 4]
```diff
@@ -267,6 +308,13 @@
                     "description": "Validation of parameters in lenient more.",
                     "hidden": true,
                     "help_text": "Allows string values that are parseable as numbers or booleans. For further information see [JSONSchema docs](https://github.com/everit-org/json-schema#lenient-mode)."
+                },
+                "pipelines_testdata_base_path": {
```
Comment (nschcolnicov) [inline nextflow_schema.json:312]: the parameter that seems to not be in use

**Review by nschcolnicov — APPROVED** [review]
Left some comments, minor things that don't affect execution or performance, we can address them before the release or after. Besides that PR looks good.

File: nextflow_schema.json (hunk 4 above)
Comment (apeltzer) [reply nextflow_schema.json:312]: Not yet no, but we will eventually all use this .

**Comment by nschcolnicov** [comment]
@apeltzer @atrigila I created a PR for the comments left in this review, not sure if we are going to merge them in this release, but to get the ball rolling in case we do:
https://github.com/nf-core/demultiplex/pull/248/files

File: modules/nf-core/custom/dumpsoftwareversions/templates/dumpsoftwareversions.py (hunk 1 above)
Comment (apeltzer) [reply modules/nf-core/custom/dumpsoftwareversions/templates/dumpsoftwareversions.py:3]: See PR https://github.com/nf-core/demultiplex/pull/249

**Pull request merged**